File: ClassTest/Student.cpp
/*
 * Student.cpp
 *
 *  Created on: 2017年3月17日
 *      Author: wangjue
 */

#include "Student.h"
#include <iostream>

double Student::rate = 0.5;

Student::Student(string n, double nv, double ev) :
		name(n), normScore(nv), examScore(ev) {
}

void Student::dispaly() const {
	std::cout << name << "\t" << normScore << "\t"
			<< normScore * rate + examScore * (1 - rate) << std::endl;
}

File: Example/src/ecjtu/soft/Student.java
package ecjtu.soft;

public class Student {
	private static int maxno = 0;
	private int no;
	String name;

	double normScore;
	double examScore;
	double finalScore;

	final double rate = 0.5;

	public Student(String name, double norm, double exam) {
		no = ++maxno;
		this.name = name;
		normScore = norm;
		examScore = exam;
		calScore();
	}

	void calScore() {
		finalScore = normScore * rate + examScore * (1 - rate);
	}

	void displayScore() {
		System.out.println("" + no + "\t" + name + "\t" + finalScore);
	}
}
